engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/libraries/DownloadLibraryTest.java

```diff
@@ -0,0 +1,162 @@
+package org.enso.interpreter.test.libraries;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import org.enso.common.RuntimeOptions;
+import org.enso.test.utils.ContextUtils;
+import org.enso.test.utils.ProjectUtils;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * This test:
+ *
+ * <ul>
+ *   <li>Creates a ZIP archive from an extension library stored in test resources.
+ *   <li>Uses this ZIP archive as a "remote" repository for libraries
+ *   <li>Creates a custom edition that declares repository with {@code jar:file} URL
+ *   <li>Creates a dummy project that uses this edition and calls a method from the extension
+ *       library
+ *   <li>Runs the project and verifies that the library was downloaded
+ * </ul>
+ */
+public final class DownloadLibraryTest {
+  @ClassRule public static final TemporaryFolder tmpFolder = new TemporaryFolder();
+  private static final String EDITION_NAME = "extra_edition";
+  private static final String LIB_DIR_RESOURCE = "download_library_test/libraries";
+  private static final String LIB_NAMESPACE = "Extension";
+  private static final String LIB_NAME = "Lib";
+  private static final String LIB_VERSION = "1.0.0";
+
+  private URL zippedLibsURL;
+  private Path editionsDir;
+  private Path projDir;
+
+  @Before
+  public void setup() throws IOException {
+    var rootDir = tmpFolder.newFolder().toPath();
+    createZippedLibrary();
+    createEditions(rootDir);
+    createProject(rootDir);
+    System.setProperty("enso.edition.path", editionsDir.toAbsolutePath().toString());
+  }
+
+  /**
+   * Creates ZIP archive from the extension library in test resources. This ZIP will be used as a
+   * "remote" repository for downloading the libraries.
+   */
+  private void createZippedLibrary() throws IOException {
+    var libDirURL = DownloadLibraryTest.class.getClassLoader().getResource(LIB_DIR_RESOURCE);
+    Path libDir;
+    try {
+      libDir = Path.of(libDirURL.toURI());
+    } catch (URISyntaxException e) {
+      throw new AssertionError("Failed to get library directory", e);
+    }
+    var zipOut = tmpFolder.newFile("Libs.zip");
+    zipDirectory(libDir, zipOut.toPath());
+    assertThat(zipOut.exists(), is(true));
+    zippedLibsURL = URI.create("jar:file:" + zipOut.toPath().toAbsolutePath() + "!/").toURL();
```

Comment:
I have also verified that `jar:https` works for a true remote ZIP archive - https://github.com/enso-org/enso/issues/14469#issuecomment-3641234414. In this test, we are just opening a connection to a local ZIP archive. There is no server and no network communication.